fan2go.yaml

```diff
@@ -46,6 +46,13 @@ fans:
     # The curve ID (defined above) that should be used to determine the
     # speed of this fan
     curve: cpu_curve
+    # (Optional) The algorithm how the target speed, determined by the curve is approached.
+    # default: pid: uses the PID control algorithm with default tuning variables
+    # simple: together with pwmChangePerSecond, fan speeds will approach target value 
+    # with the given max speed. Default is 10
+    controlAlgorithm:
+      alg: simple
```

Comment:
Let's stay with how differentiation of different types is done elsewhere within the config, meaning the type should be the "key" here.

Also, I am not very fond of the name "simple", since it doesn't really explain anything about what its doing. Your initial name for it was "linear", but I don't think thats correct here either, since this only specifies *how* the pwm values of a fan are adjusted, not *to what*, so maybe "direct" would be a better fit here?

If so, changing the parameter name to "maxPwmChangePerCycle" could better indicate that this is a modification on the "direct" behavior, which applies once per control loop cycle and puts a "cap/max" on the pwm change from one to the next cycle. Since the length of a cyce is not constant, using "Second" here might not be optimal. Although for a user using a fixed amount of time for this property regardless of how fast the cycle rate is could also be of value, but would make it more difficult to implement the control algorithm, since it would have to transform the limit to multiples of its cycle time, which would not be possible in all configuration variations.... so I would like to stay away from that, at least for the initial implementation.

Examples:

```yaml
controlAlgorithm:
    direct:
        pwmChangePerSecond: 10
```

```yaml
controlAlgorithm:
    pidLoop:
        p: 0.03
        i: 0.002
        d: 0.0005
```

Reply:
Thinking about it this might not work for types that don't have any configuration properties :thinking: But maybe we can support setting a string ("pidLoop") when using the default values and a map when not.

Reply:
I have added code to support both string values as well as a struct.